### PR TITLE
Adding Lazy Loading documentation with examples

### PR DESCRIPTION
**Pull request by guillaumebriday**
I use Vue as example but it works with React as well.

### CODE REVIEW

**Comment by guillaumebriday** [comment]
Ok I will do this changes today or tomorrow ! Thank you 👌

**Comment by guillaumebriday** [comment]
Ok @jakeNiemiec, is it better now ? 😄

File: docs/es6.md [hunk 1]
```diff
@@ -26,6 +25,15 @@ import "core-js/stable";
...
 
+## Lazy Loading
+
+[Webpack Lazy Loading](https://webpack.js.org/guides/lazy-loading/) is avaible out of the box with split chunks.
+
+To make it works, you need Webpack and Webpacker 4. Then enable `SplitChunks` as it is explained in [docs/webpack](docs/webpack.md).
+
+Dynamic import and [Code splitting](https://webpack.js.org/guides/code-splitting/) are actually a Webpack features that just work with Webpacker out of the box.
+
+Don't forget that you can use [magic comments](https://webpack.js.org/api/module-methods/#magic-comments) to control your chunks options.
```
Comment (jakeNiemiec) [inline docs/es6.md]: Here Is what I would use for this section. Hope you don't mind, I run into this topic in the issues often. What do you think?

<details><summary>raw MD</summary>

```
## Dynamic/Lazy Chunk Loading
[Dynamic code splitting](https://webpack.js.org/guides/code-splitting#dynamic-imports) enables you to conditionally request/run only the JS that you need. For example, if your site has a `searchBarComponent` on every page, you can reduce the page overhead by deferring the request for the `searchBarComponent` code until after the page has loaded,  until the user has scrolled it into view, or until the user has clicked on an element.
``js
 function loadSearchBarComponent() {
   return import(/* webpackChunkName: "searchBarComponent" */ './pathTo/searchBarComponent')
 }
``
The comment you see above (`/* webpackChunkName */`) is not arbitrary, it is one of webpacks [magic comments](https://webpack.js.org/api/module-methods/#magic-comments). They can be used to fine-tune `import()` with settings such as `defer` or `prefetch`. 

**Warning**: You should not attempt to dynamically load anything from your `packs/` folder. Instead, try to make your `pack` scripts a hub from which you dynamically load `non-pack` scripts.

- [Docs for using magic comments](https://webpack.js.org/api/module-methods/#magic-comments)
- [Docs for configuring `splitChunks` in webpacker](https://github.com/rails/webpacker/blob/master/docs/webpack.md#add-splitchunks-webpack-v4).
- [Docs for using dynamic `import()`](https://webpack.js.org/guides/code-splitting#dynamic-imports).

```

</details>

<hr>

## Dynamic/Lazy Chunk Loading
[Dynamic code splitting](https://webpack.js.org/guides/code-splitting#dynamic-imports) enables you to conditionally request/run only the JS that you need. For example, if your site has a `searchBarComponent` on every page, you can reduce the page overhead by deferring the request for the `searchBarComponent` code until after the page has loaded,  until the user has scrolled it into view, or until the user has clicked on an element.
```js
 function loadSearchBarComponent() {
   return import(/* webpackChunkName: "searchBarComponent" */ './pathTo/searchBarComponent')
 }
```
The comment you see above (`/* webpackChunkName */`) is not arbitrary, it is one of webpacks [magic comments](https://webpack.js.org/api/module-methods/#magic-comments). They can be used to fine-tune `import()` with settings such as `defer` or `prefetch`. 

**Warning**: You should not attempt to dynamically load anything from your `packs/` folder. Instead, try to make your `pack` scripts a hub from which you dynamically load `non-pack` scripts.

- [Docs for using magic comments](https://webpack.js.org/api/module-methods/#magic-comments)
- [Docs for configuring `splitChunks` in webpacker](https://github.com/rails/webpacker/blob/master/docs/webpack.md#add-splitchunks-webpack-v4).
- [Docs for using dynamic `import()`](https://webpack.js.org/guides/code-splitting#dynamic-imports).

**Comment by guillaumebriday** [comment]
Ok it is good this time @jakeNiemiec ?

**Pull request merged**